ssg/controls.py

```diff
@@ -198,13 +201,32 @@ def _get_policy(self, policy_id):
     def get_all_controls_of_level(self, policy_id, level_id):
         policy = self._get_policy(policy_id)
         levels = policy.get_level_with_ancestors(level_id)
-        level_ids = set([lv.id for lv in levels])
-
         all_policy_controls = self.get_all_controls(policy_id)
         eligible_controls = []
-        for c in all_policy_controls:
-            if len(level_ids.intersection(c.levels)) > 0:
-                eligible_controls.append(c)
+        defined_variables = []
+        # we will go level by level, from top to bottom
+        # this is done to enable overriding of variables by higher levels
+        for lv in levels.keys():
+            for c in all_policy_controls:
+                if lv.id in c.levels:
```

Comment:
This `if` can be simplified by `if lv.id not in c.levels: continue` - one level of indentation less.

The following code is not complex, but it needs extraction into meaningfully named private functions. Right now, it is very difficult to guess what are we doing why.